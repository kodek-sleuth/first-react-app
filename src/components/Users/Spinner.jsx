import React from 'react';

const Spinner = () => {
    return (
        <div>
            <img src="https://loading.io/spinners/rolling/lg.curve-bars-loading-indicator.gif" alt="" style={{ margin: 'auto', display: 'block', width: '50px' }}/>
        </div>
    )
}

export default Spinner;
