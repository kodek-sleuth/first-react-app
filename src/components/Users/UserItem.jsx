import React from "react"; 
import { Link } from 'react-router-dom';

const UserItem = (props) => {
  const { login, avatar_url } = props.user;
  return (
    <div>
      <div className="card">
        <img src={avatar_url} alt="" className="avatar"/>
        <h3>{login}</h3>
        <Link to={`/user/${login}`}><button className="more">More</button></Link>
      </div>
    </div>
  );
}

export default UserItem;