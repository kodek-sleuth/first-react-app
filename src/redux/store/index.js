import { createStore, applyMiddleware } from 'redux';
import reducer from '../reducers';
import thunk from 'redux-thunk';
import { composeWithDevTools } from 'redux-devtools-extension'

// Redux Thunk is a middleware that letsyou call an action creator that returns a function instead of an action object

const store = createStore(reducer, composeWithDevTools(applyMiddleware(thunk)));

export default store;