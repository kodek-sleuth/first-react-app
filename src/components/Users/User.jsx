import React, { useEffect } from 'react'; 
import { connect } from 'react-redux';
import { Link, Redirect } from 'react-router-dom';
import { getSingleUser } from '../../redux/actions/actionCreators';
import Spinner from '../Users/Spinner.jsx';

const User = props => {
  useEffect(() => {
    const username = props.match.params.login;
    props.getSingleUser(username)
  }, [])

  const redirectToProfile = () => {
    window.location = props.user.blog
  }

  const { name, avatar_url } = props.user;
  if (props.loading){
    return <Spinner />
  }
  return (
    <div className="card">
      <img src={avatar_url} alt="" className="avatar"/>
      <h3>{name}</h3>
      <button className="more" onClick={redirectToProfile}>Profile</button>
    </div>
  );
}

const mapStateToProps = state => ({
  loading: state.loading,
  user: state.user
})

export default connect(mapStateToProps, {getSingleUser})(User)