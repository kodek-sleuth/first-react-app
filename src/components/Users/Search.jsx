import React, { Component } from 'react';
import { connect } from 'react-redux';
import { searchUsers } from '../../redux/actions/actionCreators';
import UserItem from './UserItem.jsx';
import Spinner from '../Users/Spinner.jsx';

export class Search extends Component {
    state = {
        text: ''
    }

    onChange = (e) => {
        console.log(this.props);
        this.setState({ text: e.target.value })
    }

    onSubmit = (e) => {
        e.preventDefault();
        this.props.searchUsers(this.state.text)
    }

    render() {
        if (this.props.loading){
            return <Spinner />
        }
        return (
            <div>
                <form style={{ marginTop: '100px', marginLeft: '100px' }} onSubmit={this.onSubmit}>
                    <input type="search" placeholder="search..." className="formControl" value={this.state.text} onChange={this.onChange} />
                    <input type="submit" value="Search" className="formControlBtn" />
                </form>
                <div className="container">
                    {
                        this.props.users.map(user => 
                        <div key={user.id}>
                            <UserItem user={user} />
                        </div>)
                    }
                </div>
            </div>
        )
    }
}

const mapStateToProps = state => ({
    loading: state.loading,
    users: state.users
})

export default connect(mapStateToProps, {searchUsers})(Search)
