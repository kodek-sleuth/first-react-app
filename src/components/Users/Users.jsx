import React, { useEffect } from 'react';
import { connect } from 'react-redux';
import { getUsers } from '../../redux/actions/actionCreators';
import UserItem from './UserItem.jsx';
import Spinner from '../Users/Spinner.jsx';

export const Users = props => {
    useEffect(() => {
        props.getUsers();
    }, [])

    if (props.loading){
        return <Spinner />
    }

    return(
        <div className="container">
            {props.users.map(user => 
                <div key={user.id}>
                    <UserItem user={user}/>
                </div>
            )}
        </div>
    )
}

const mapStateToProps = state => ({
    users: state.users,
    loading: state.loading
})

export default connect(mapStateToProps, {getUsers})(Users)
