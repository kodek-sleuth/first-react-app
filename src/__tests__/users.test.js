import React from 'react';
import { configure, shallow, mount } from 'enzyme';
import { create } from "react-test-renderer";
import Adapter from 'enzyme-adapter-react-16';
import store from  '../redux/store';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-dom';
import { Users } from '../components/Users/Users';
import UserItem from '../components/Users/UserItem';
import Spinner from '../components/Users/Spinner';
import { User } from '../components/Users/User';

configure({ adapter: new Adapter() });

const props = {
    getUsers:jest.fn(),
    users: [{
        login: 'mojombo',
        avatar_url: "https://avatars0.githubusercontent.com/u/1?v=4"
    }],
    loading: true || false
}

const userProps = {
    getSingleUser:jest.fn(),
    user: [{
        login: 'mojombo',
        name: 'mojombo mark',
        blog: "https://avatars0.githubusercontent.com/u/1?v=4"
    }],
    loading: false
}

const userItemProps = {
    user: [{
        login: 'mojombo',
        avatar_url: "https://avatars0.githubusercontent.com/u/1?v=4"
    }],
    loading: false
}



describe('<Users />', () => {
    it('should return all users on the page', () => {
        const component = shallow(<Users {...props}/>);
        expect(component).toMatchSnapshot();
    });
    it('should display a container', () => {
        const component = shallow(<UserItem {...userItemProps} />);
        expect(component.find('div')).toHaveLength(2)
    });
});

describe('<Spinner />', () => {
    it('should display spinner', () => {
        const component = shallow(<Spinner />);
        expect(component.find('div')).toHaveLength(1)
    });
})

describe('<User />', () => {
    it('should return a user', () => {
        const component = shallow(<User {...userProps} />);
        expect(component).toMatchSnapshot();
    });
})
