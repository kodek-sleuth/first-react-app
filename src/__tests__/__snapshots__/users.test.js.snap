// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<User /> should return a user 1`] = `ShallowWrapper {}`;

exports[`<Users /> should return all users on the page 1`] = `ShallowWrapper {}`;
